esp/esp/users/models/__init__.py

```diff
@@ -1354,6 +1362,7 @@ class TeacherInfo(models.Model, CustomFormsLinkModel):
 
     user = AjaxForeignKey(ESPUser, blank=True, null=True)
     graduation_year = models.CharField(max_length=4, blank=True, null=True)
+    affiliation = models.CharField(max_length=100, blank=True, null=True)
```

Comment:
It doesn't work with the DropdownOtherField, because in case the user selects "Other", it stores the field as a colon-separated string "Other:Whatever" where "Whatever" is the text that the user typed into the other box.  So there aren't actually a finite number of choices to list.